esmvalcore/_recipe/recipe.py

```diff
@@ -641,12 +642,32 @@ def _update_extract_shape(settings, session):
     if 'extract_shape' in settings:
         shapefile = settings['extract_shape'].get('shapefile')
         if shapefile:
-            if not os.path.exists(shapefile):
-                shapefile = os.path.join(
-                    session['auxiliary_data_dir'],
-                    shapefile,
+            shapefile = Path(shapefile)
+            abs_shapefile = shapefile
+
+            # If shapefile not found, first, try path relative to
+            # auxiliary_data_dir
+            if not abs_shapefile.exists():
+                abs_shapefile = session['auxiliary_data_dir'] / shapefile
+
+            # Second, try path relative to esmvalcore/preprocessor/shapefiles/
+            if not abs_shapefile.exists():
+                abs_shapefile = (
+                    Path(esmvalcore.preprocessor.__file__).parent /
+                    'shapefiles' /
+                    shapefile
                 )
-                settings['extract_shape']['shapefile'] = shapefile
+
+            # As final resort, add suffix '.shp' and try path relative to
+            # esmvalcore/preprocessor/shapefiles/ again
+            if not abs_shapefile.exists():
+                abs_shapefile = (
+                    Path(esmvalcore.preprocessor.__file__).parent /
+                    'shapefiles' /
+                    shapefile.with_suffix('.shp')
+                )
+
+            settings['extract_shape']['shapefile'] = abs_shapefile
```

Comment:
I would argue a DEBUG message would be nice to have per each of these looked into locations, that'd help the user see if they actually have some sort of shapefile, but it's not the one the code is looking for

Reply:
Good idea, done in https://github.com/ESMValGroup/ESMValCore/pull/2008/commits/fb1dd77340bbf906089b06d445b53fff05c5268f

Reply:
cheers Manu! I'll get back to the rest of the review later today, a bit of a lax schedule for me these days with the Big C19 :grin: